### DISCUSSION TITLE
Add a textbox on the detail page

### DISCUSSION THREAD

**Discussion by Rodien** [openstore-ecommerce/OpenStore#202] [Q&A]
Hi Guys,

I want to add a product length field on the detail page. 

Use case:
Users should be able to input the length of a product they want to buy.
The value should be displayed in the cart and on the invoice. 

How I'm going about it:

`@TextBox(info, "genxml/textbox/unitqty", "class='lengthquantity selectedaddqty'", "1")`
![image](https://user-images.githubusercontent.com/50739829/137309843-66917d93-27ba-49bd-99b0-20bb16d91f1b.png)

I put the code above on the detailpage but I do not know if that is the correct way to go about it. 

How can I display the value on the cart list and on the invoice?

Or is there a better way?

@leedavi @DNNMonster

**Comment by DNNMonster** [comment]
It looks to be a good candidate for being a product option.  

First I'd check that out to see if it will work for you.  That should also render on your cart list and invoice without needing to do any coding or template modifications.  Just a small modification to your product details template.

If you create your length option but don't add any values....

Then try adding the ProductOption token on your display just below the ProductOptions token and you should find things work as desired.  

`@if (@product.Options.Count >= 1)
                    {
                        <div class="options mt-3">@ProductOptions(product)</div>
                        @ProductOption(product)
                    }`

**Comment by leedavi** [comment]
You can also add it to \Themes\config\default\productfields.cshtml

Also, put you changes at portal level, so they don;t get overwritten on upgrade.
